### DISCUSSION TITLE
Welcome to flake8-async Discussions!

### DISCUSSION THREAD

**Discussion by cooperlees** [cooperlees/flake8-async#3] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by cooperlees** [comment]
@Zac-HD - With your release CI - Do you just rely on a merge / commit to increment the version in order to trigger a release?
- Does PyPI block the upload of the same version? I would imagine so.

**Reply by Zac-HD** [reply]
More precisely, [our `twine upload --skip-existing`](https://github.com/cooperlees/flake8-async/blob/ce0237bbc3b81578249d282bfeb25da88cbb2b90/.github/workflows/ci.yml#L62) will... skip uploading existing versions.  So every CI run on master checks if it's got a new version number, and cuts a release if it does.

`test_versions.py` then exists to make sure the build doesn't get that far unless it looks like you meant it, with a changelog declaring today's date and the same version number.  It's, ah, more expedient than Hypothesis' system, but way less code to implement and I find it works nicely for projects where you're unlikely to have multiple PRs around at a time and don't need to worry about merge conflicts in the changelog.
